### PR TITLE
Add exporter documentation

### LINKED ISSUE #1181: Document exporters
## Criterias

- A new exporter section is added to the documentation.
- Section explains the how and why of exporters, as well as current limitations.
- Contains examples on how to write a basic exporter.

### Notes

Blocked by #1173

### PR DESCRIPTION
**Pull request by npepinpe**
Adds two pages: basic information on exporters (roles and so on), and a reference page for more in-depth implementation details.

Note that this does not include a guide to write your own exporter, I've outsourced to this another issue since, while nice, it's not necessary at the moment. See #1473.

closes #1181

### CODE REVIEW

File: docs/src/basics/exporters.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+You can find a reference exporter implementation in the form of the Zeebe-maintained
+[ElasticSearch exporter](https://github.com/zeebe-io/zeebe/tree/{{commit}}/exporters/elasticsearch-exporter).
+
+For more information, you can read the [reference information page](reference/exporters.html).
+
+> **Note**: it is not recommended to use exporters for orchestration purposes, as
+> that is what workflows are for.
+
+## Usage
```
Comment (ThorbenLindhauer) [inline docs/src/basics/exporters.md]: I think this section fits better into the Reference -> Exporters chapter. The Basics section is meant to explain the principles of Zeebe to people (maybe non-technical) who are new to it. As a user who has understood these principles and who wants to build something with Zeebe, I should not have to go back to the Basics section.

File: docs/src/basics/exporters.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+Zeebe will only load exporters which are configured through the main Zeebe TOML
```
Comment (ThorbenLindhauer) [inline docs/src/basics/exporters.md]: We could link to Operations -> toml config here.

File: docs/src/basics/exporters.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+configuration file.
+
+Exporters are configured as TOML maps, with a standard preamble describing the
+exporter:
+
+```toml
+[[exporters]]
+id = "myExporter"
+className = "com.myCompany.MyExporter"
```
Comment (ThorbenLindhauer) [inline docs/src/basics/exporters.md]: It's not clear to me where I would have to put the class to make it work. I guess it is Zeebe's classpath. We should mention that explicitly.

File: docs/src/basics/exporters.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+```
+
+Optionally, an exporter can be loaded through an external JAR file, simply by
+pointing to it through the configuration:
+
+```toml
+[[exporters]]
+id = "myExporter"
+className = "com.myCompany.MyExporter"
+jarPath = "exporters/com.myCompany.exporters.jar"
```
Comment (ThorbenLindhauer) [inline docs/src/basics/exporters.md]: To contrast this with the configuration before, we could mention explicitly that this JAR does not have to be on Zeebe's classpath.

File: docs/src/basics/exporters.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+Exporters are configured as TOML maps, with a standard preamble describing the
+exporter:
+
+```toml
+[[exporters]]
+id = "myExporter"
+className = "com.myCompany.MyExporter"
+```
+
+Optionally, an exporter can be loaded through an external JAR file, simply by
```
Comment (ThorbenLindhauer) [inline docs/src/basics/exporters.md]: As a user who wants to implement an exporter, I would like to know how I can manage my exporter's depencencies. e.g. does the exporter jar have to be a fat jar?

File: docs/src/basics/exporters.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+under the label `[exporters.args]`:
+
+```toml
+[[exporters]]
+id = "myExporter"
+className = "com.myCompany.MyExporter"
+jarPath = "exporters/com.myCompany.exporters.jar"
+  [exporters.args]
+  username = "mmustermann"
+  host = "udp://myCompany.com:8080"
```
Comment (ThorbenLindhauer) [inline docs/src/basics/exporters.md]: Here I asked myself how I would access the configuration. I think that also gets clearer if this section goes into Reference -> Exporters.

File: docs/src/basics/exporters.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+of persisting data indefinitely.
+
+Once data is not needed by Zeebe itself anymore, it will query its exporters to
+know if it can be safely deleted, and if so, will permanently erase it, thereby
+reducing disk usage.
+
+Therefore, if no exporters are configured at all, then Zeebe will automatically
+erase data when it is not necessary anymore. If you need historical data, an exporter is thus necessary.
+
+## Performance impact
```
Comment (ThorbenLindhauer) [inline docs/src/basics/exporters.md]: This should probably also go to Reference -> Exporters

File: docs/src/basics/exporters.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+The main impact exporters have on a Zeebe cluster is that they remove the burden
+of persisting data indefinitely.
+
+Once data is not needed by Zeebe itself anymore, it will query its exporters to
+know if it can be safely deleted, and if so, will permanently erase it, thereby
+reducing disk usage.
+
+Therefore, if no exporters are configured at all, then Zeebe will automatically
+erase data when it is not necessary anymore. If you need historical data, an exporter is thus necessary.
```
Comment (ThorbenLindhauer) [inline docs/src/basics/exporters.md]: This is a key point for this Basics section for me. `No exporter => no history` is an important concept.

File: docs/src/reference/exporters.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+sequence of records whose position is monotonically increasing. This position is
+set by the exporter itself once it can guarantee a record has been successfully
+updated.
+
+> **Note:** although Zeebe tries to reduce the amount of duplicate records an
+> exporter has to handle, it is likely that it will have to; therefore, it is
+> necessary that export operations be idempotent. This can be implemented either in
+> the exporter itself, but if it exports to an external system, it is recommended
+> that you perform deduplication there to reduce the load on Zeebe itself. Refer
+> to the exporter specific documentation for how this is meant to be achieved.
```
Comment (ThorbenLindhauer) [inline docs/src/reference/exporters.md]: I find the last sentence confusing, as I am already reading exporter-specific documentation here :) Also, it makes assumptions how people document their own code.

Maybe we can rather assume that the user who configures the exporter is the same user who implements it (or at least has the end-to-end knowledge). Then we could simply drop this sentence.

**Review by ThorbenLindhauer — CHANGES_REQUESTED** [review]
I like the scope of the documentation and the clarity of your explanations. I think people will benefit from a full working example, so I'm looking forward to #1473 :)

I added some comments where we could move stuff around or make things more clear. Let's discuss this if you disagree with the points or would like clarification. I also removed some in my opinion unnecessary statements: https://github.com/zeebe-io/zeebe/pull/1474/commits/0f20377423f7fb03e832bf1371b025c2dce49add.

File: docs/src/basics/exporters.md (hunk 5 above)
Comment (npepinpe) [reply docs/src/basics/exporters.md]: I will mention it in the reference

File: docs/src/basics/exporters.md (hunk 2 above)
Comment (npepinpe) [reply docs/src/basics/exporters.md]: Leave sentence and remove details

File: docs/src/basics/exporters.md (hunk 7 above)
Comment (npepinpe) [reply docs/src/basics/exporters.md]: Move to reference

File: docs/src/basics/exporters.md (hunk 8 above)
Comment (npepinpe) [reply docs/src/basics/exporters.md]: Keep in basics but emphasize (blockquote)

File: docs/src/reference/exporters.md (hunk 9 above)
Comment (npepinpe) [reply docs/src/reference/exporters.md]: To remove

**Comment by npepinpe** [comment]
I worked in the changes we discuss, and I'll start on the Elasticsearch Exporter documentation

**Review by ThorbenLindhauer — APPROVED** [review]

**Pull request merged**